src/leap/bitmask/logs/utils.py

```diff
@@ -1,80 +1,57 @@
 import logging
 import sys
 
+from leap.bitmask.config import flags
 from leap.bitmask.logs import LOG_FORMAT
 from leap.bitmask.logs.log_silencer import SelectiveSilencerFilter
-from leap.bitmask.logs.leap_log_handler import LeapLogHandler
+from leap.bitmask.logs.safezmqhandler import SafeZMQHandler
 from leap.bitmask.logs.streamtologger import StreamToLogger
 from leap.bitmask.platform_init import IS_WIN
 
+import logbook
+from logbook.more import ColorizedStderrHandler
 
-def create_logger(debug=False, logfile=None, replace_stdout=True):
-    """
-    Create the logger and attach the handlers.
-
-    :param debug: the level of the messages that we should log
-    :type debug: bool
-    :param logfile: the file name of where we should to save the logs
-    :type logfile: str
-    :return: the new logger with the attached handlers.
-    :rtype: logging.Logger
-    """
-    # TODO: get severity from command line args
-    if debug:
-        level = logging.DEBUG
-    else:
-        level = logging.WARNING
 
-    # Create logger and formatter
-    logger = logging.getLogger(name='leap')
-    logger.setLevel(level)
-    formatter = logging.Formatter(LOG_FORMAT)
+def get_logger():
+    level = logbook.WARNING
+    if flags.DEBUG:
+        level = logbook.NOTSET
 
-    # Console handler
-    try:
-        import coloredlogs
-        console = coloredlogs.ColoredStreamHandler(level=level)
-    except ImportError:
-        console = logging.StreamHandler()
-        console.setLevel(level)
-        console.setFormatter(formatter)
-        using_coloredlog = False
-    else:
-        using_coloredlog = True
-
-    if using_coloredlog:
-        replace_stdout = False
+    # This handler consumes logs not handled by the others
+    null_handler = logbook.NullHandler(bubble=False)
+    null_handler.push_application()
 
     silencer = SelectiveSilencerFilter()
-    console.addFilter(silencer)
-    logger.addHandler(console)
-    logger.debug('Console handler plugged!')
 
-    # LEAP custom handler
-    leap_handler = LeapLogHandler()
-    leap_handler.setLevel(level)
-    leap_handler.addFilter(silencer)
-    logger.addHandler(leap_handler)
-    logger.debug('Leap handler plugged!')
+    zmq_handler = SafeZMQHandler('tcp://127.0.0.1:5000', multi=True,
+                                 level=level, filter=silencer.filter)
```

Comment:
Maybe i'm getting the wrong picture, but it seems to me that we have one ZMQ connection per thread that uses a logger returned by `get_logger()`.